base/applications/drwtsn32/stacktrace.cpp

```diff
@@ -59,23 +61,28 @@ void PrintStackBacktrace(FILE* output, DumpData& data, ThreadData& thread)
     char buf[sizeof(SYMBOL_INFO) + STACKWALK_MAX_NAMELEN] = {0};
     SYMBOL_INFO* sym = (SYMBOL_INFO *)buf;
     IMAGEHLP_MODULE64 Module = { 0 };
+    LONG RecursionDepth = 0;
     sym->SizeOfStruct = sizeof(sym);
 
-    /* FIXME: Disasm function! */
+    /* FIXME: dump x bytes at EIP here + disasm it! */
 
     xfprintf(output, NEWLINE "*----> Stack Back Trace <----*" NEWLINE NEWLINE);
     bool first = true;
-    ULONG_PTR LastFrame = StackFrame.AddrFrame.Offset - 8;
-    while(StackWalk64(MachineType, data.ProcessHandle, thread.Handle, &StackFrame, &thread.Context,
-                         NULL, SymFunctionTableAccess64, SymGetModuleBase64, NULL))
+    while (StackWalk64(MachineType, data.ProcessHandle, thread.Handle, &StackFrame, &thread.Context,
+                       NULL, SymFunctionTableAccess64, SymGetModuleBase64, NULL))
     {
-        if (!StackFrame.AddrPC.Offset)
-            break;
-
-        if (LastFrame >= StackFrame.AddrFrame.Offset)
```

Comment:
Dropping this check intentionally?

Reply:
Yes.

Reply:
The previous checks were too strict, and thus we were not showing a callstack when it was still possible to show something.